plugins/kubernetes-cluster/src/Router.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { Route, Routes } from 'react-router-dom';
+import { MissingAnnotationEmptyState } from '@backstage/core-components';
+import { ANNOTATION_KUBERNETES_API_SERVER } from '@backstage/plugin-kubernetes-common';
+import { KubernetesClusterContent } from './components/KubernetesClusterContent';
+
+/**
+ *
+ *
+ * @public
+ */
+export const isKubernetesClusterAvailable = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.[ANNOTATION_KUBERNETES_API_SERVER]);
```

Comment:
This is very similar to -- but slightly different from -- the filter used by the `CatalogClusterLocator` to actually surface clusters:

https://github.com/backstage/backstage/blob/b74f157839a2a3f6b6051b58c27c9c360300bc75/plugins/kubernetes-backend/src/cluster-locator/CatalogClusterLocator.ts#L45-L51

Ideally this predicate would be extracted to some common location and used by both plugins. Doesn't need to happen before merging this, but it did get me into trouble while testing with a cluster like this:

```yaml
apiVersion: backstage.io/v1alpha1
kind: Resource
metadata: 
  name: aks-cluster
  annotations: 
    kubernetes.io/api-server: https://my.aks.url:443
    kubernetes.io/auth-provider: aks
    kubernetes.io/skip-metrics-lookup: 'true'
    kubernetes.io/skip-tls-verify: 'true'
spec: 
  owner: user:guest
  type: kubernetes-cluster
```

Without the `kubernetes.io/api-server-certificate-authority` annotation, the 'Kubernetes Cluster' tab still appeared on the entity page for `aks-cluster` despite it not being recognized by the cluster locator.